src/message/headers/TopicMessageHeader.js

```diff
@@ -38,15 +38,30 @@ export default class TopicMessageHeader extends React.PureComponent {
     doNarrow(topicNarrow(stream, topic), itemId);
   }
 
+  topicText() {
+    const { topic, isTopicNarrow } = this.props;
+    if (isTopicNarrow) {
```

Comment:
The end result looks good.
The only reservation I have is about this code.
Do we want to have a multiline topic if the narrow is in a topic?
I don't think we want to treat these differently.